test/cppsuite/src/main/thread_worker.cpp

```diff
@@ -210,6 +210,47 @@ thread_worker::remove(scoped_cursor &cursor, uint64_t collection_id, const std::
     return (ret == 0);
 }
 
+bool
+thread_worker::truncate(uint64_t collection_id, std::optional<std::string> start_key,
+  std::optional<std::string> stop_key, const std::string &config)
+{
+    WT_DECL_RET;
+
+    wt_timestamp_t ts = tsm->get_next_ts();
+    ret = txn.set_commit_timestamp(ts);
+    testutil_assert(ret == 0 || ret == EINVAL);
+    if (ret != 0) {
+        txn.set_needs_rollback(true);
+        return (false);
+    }
+
+    const std::string coll_name = database::build_collection_name(collection_id).c_str();
+
+    scoped_cursor start_cursor = session.open_scoped_cursor(coll_name);
+    scoped_cursor stop_cursor = session.open_scoped_cursor(coll_name);
+    if (start_key.has_value())
+        start_cursor->set_key(start_cursor.get(), start_key.value().c_str());
+
+    if (stop_key.has_value())
+        stop_cursor->set_key(stop_cursor.get(), stop_key.value().c_str());
+
+    ret = session->truncate(session.get(),
+      (start_key.has_value() || stop_key.has_value()) ? nullptr : coll_name.c_str(),
```

Comment:
Why does the `uri` arg depends on `start_key` or `stop_key`?

Reply:
There is a check in truncate which checks for these, I can't remember the exact check but will try to find it and get back you.

Reply:
@etienneptl In the `__session_truncate` function on line 1568 there's a check that if the URI is specified, we don't need a start/stop, if start/stop is specified, we don't need a URI.

Reply:
The start/stop cursors will only exist/be non-null if the start_key or stop_key has a value. Let me know if that doesn't make sense.

Reply:
I did not know that, thank you for the info! It makes sense.